### PR TITLE
attic-client,attic-server: 0-unstable-2024-02-08 -> 0-unstable-2024-08-01

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

[Required to build with Rust 1.80][1].

Note: I'm pretty sure the labelling of the previous version as 0.1.0 was a mistake — the commit message referred to an unstable version rather than 0.1.0, and there's no 0.1.0 tag upstream.

[1]: https://github.com/rust-lang/rust/issues/127343

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/at/attic-client/package.nix [hunk 1]
```diff
@@ -11,13 +11,13 @@
 }:
 rustPlatform.buildRustPackage {
   pname = "attic";
-  version = "0.1.0";
+  version = "0-unstable-2024-08-01";
```
Comment (aciceri) [inline pkgs/by-name/at/attic-client/package.nix:14]: Why the `0-` prefix?
Anyway in the upstream derivation the `version` is `0.1.0`, and also in all the `Cargo.toml`s it says `0.1.0`. Consequently also `attic --version` reports `0.1.0`.

[Here](https://github.com/NixOS/nixpkgs/pull/274481#issuecomment-1994062002) my message about it.

Said this, I was expecting a release soon when I did that change, so probably renaming to `unstable-<date>` makes more sense now.

Comment (alyssais) [reply pkgs/by-name/at/attic-client/package.nix:14]: > Why the `0-` prefix?

https://github.com/NixOS/nixpkgs/blob/102b1152a6ea69f4be9d66bde3cb3141d5642b6c/pkgs/README.md#package-naming

Comment (aciceri) [reply pkgs/by-name/at/attic-client/package.nix:14]: Oh I didn't know, thanks :)

**Review by aciceri — APPROVED** [review]

**Comment by alyssais** [comment]
@OfBorg build attic-client attic-server

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**